search/tests.py

```diff
@@ -193,6 +195,40 @@ def test_search_page_all_params(self):
         self.assertContains(response, "Min Price: 500")
         self.assertContains(response, "Number of Bedroom: 4")
 
+    # @mock.patch("search.views.fetch_craigslist_housing", fetch_craigslist_housing)
+    # @mock.patch("external.nyc311.fetch.fetch_311_data", fetch_311_data)
+    # @mock.patch("external.res.fetch.fetch_res_data", fetch_res_data)
+    # def test_search_page_matching_apartments(self):
+    #     # create a location
+    #     city = "Brooklyn"
+    #     state = "NY"
+    #     address = "1234 Coney Island Avenue"
+    #     zipcode = 11218
+    #     # using get_or_create avoids race condition
+    #     loc = Location.objects.get_or_create(
+    #         city=city, state=state, address=address, zipcode=zipcode
+    #     )[0]
+    #
+    #     # create an apartment and link it to that location
+    #     suite_num = "18C"
+    #     number_of_bed = 4
+    #     image = "Images/System_Data_Flow_Diagram.png"
+    #     rent_price = 1500
+    #     apt = Apartment.objects.create(
+    #         suite_num=suite_num,
+    #         number_of_bed=number_of_bed,
+    #         image=image,
+    #         rent_price=rent_price,
+    #         location=loc,
+    #     )
+    #     apt.save()
+    #
+    #     response = self.client.get(
+    #         "/search/?query=11218&min_price=500&max_price=2000&bed_num=4"
```

Comment:
you may need to also mock the `normalize_us_address` function so it returns the search terms that you want (and also doesn't make a call to geocode. you can use `@mock.patch("search.views.normalize_us_address")`. For reference you can take a look at lines 224-246 in `search/tests.py` in my branch feature/apa-rented

Reply:
Thank you for the reference! It worked. Also, changed the formatting to 2 rows and 3 cols and added a box shadow.

Reply:
excellent, looks much more user friendly now